include/deal.II/fe/fe_interface.h

```diff
@@ -0,0 +1,787 @@
+  get_quadrature_points() const
+  {
+    return internal_fe_face_values.get_quadrature_points();
+  }
+
+  /**
+   * Return the update flags set.
+   */
+  const UpdateFlags
+  get_update_flags() const
+  {
+    return internal_fe_face_values.get_update_flags();
+  }
+
+
+  /**
+   * Return the number of DoFs on the current facet. This is the sum of the
+   * number of DoFs on the two adjacent cells.
+   */
+  unsigned
+  n_interface_dofs() const
+  {
+    return n_dofs_fe + (is_boundary_facet() ? 0 : n_dofs_fe_neighbor);
+  }
+
+  /**
+   * Return if the current facet, which is set by reinit(), is an internal face
+   * with two adjacent cells or a boundary face.
+   */
+  bool
+  is_boundary_facet() const
+  {
+    return fe_face_values_neighbor == nullptr;
+  }
+
+  /**
+   * Return the set of joint DoF indices. This includes indices from both cells.
+   */
+  std::vector<types::global_dof_index>
+  get_interface_dof_indices() const
+  {
+    Assert(interface_dof_indices.size() == n_interface_dofs(),
+           ExcInternalError());
+    return interface_dof_indices;
+  }
+
+  /**
+   * Translate a local interface_dof_index to a local dof_index of the
+   * underlying FEFaceValues.
+   */
+  unsigned int
+  interface_dof_index_to_fe_dof_index(unsigned int interface_dof_index) const
+  {
+    AssertIndexRange(interface_dof_index, n_interface_dofs());
+    return (interface_dof_index >= n_dofs_fe) ?
+             (interface_dof_index - n_dofs_fe) :
+             interface_dof_index;
+  }
+
+  /**
+   * For a given interface dof index return whether it belongs to the cell (0)
+   * or the neighbor (1).
+   */
+  unsigned int
+  interface_dof_index_to_fe_index(const unsigned int interface_dof_index) const
+  {
+    AssertIndexRange(interface_dof_index, n_interface_dofs());
+    return (interface_dof_index < n_dofs_fe) ? 0 : 1;
+  }
+
+  /**
+   * Convert an FiniteElement index (0=cell, 1=neighbor) and dof index @p face_dof_index
+   * into a interface DoF index.
+   *
+   * The inverse of this operation is done with
+   * interface_dof_index_to_fe_index() and interface_dof_idx_to_fe_dof_idx().
+   */
+  unsigned int
+  interface_dof_index(const unsigned int fe_index,
+                      const unsigned int face_dof_index) const
+  {
+    Assert(fe_index <= 1, ExcMessage("fe_index should be 0 or 1"));
+    if (is_boundary_facet())
+      Assert(fe_index == 0,
+             ExcMessage("A boundary facet only has FE index 0."));
+
+    if (fe_index == 0)
+      {
+        Assert(face_dof_index < n_dofs_fe, ExcMessage("invalid face_dof_idx"));
+        return face_dof_index;
+      }
+    else if (fe_index == 1)
+      {
+        Assert(face_dof_index < n_dofs_fe_neighbor,
+               ExcMessage("invalid face_dof_idx"));
+        return face_dof_index + n_dofs_fe;
+      }
+    // return something invalid:
+    return -1;
+  }
+
+  /**
+   * Return the FEFaceValue object of the cell or the neighboring cell
```

Comment:
Can you document that the values of `cell_or_neighbor` can only be zero (this cell) or one (neighbor cell)?